spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/source/ScanTestBase.java

```diff
@@ -86,6 +87,20 @@ protected void writeAndValidate(Schema schema) throws IOException {
     writeAndValidate(schema, schema);
   }
 
+  private boolean checkVersion(List<Types.NestedField> fields) {
```

Comment:
I don't think this change is necessary because format versions are checked when creating table metadata. `TableMetadata` will reject a schema v3 types if the format version is < 3.

Reply:
Ah, I see we just changed this for Spark 4.0: https://github.com/apache/iceberg/blame/f7e6a2775bd5a9e202465e6281c56d8c41fd7f0e/spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/source/ScanTestBase.java